jujugui/static/gui/src/app/components/store/store.js

```diff
@@ -23,33 +23,820 @@ YUI.add('store', function() {
+
+      @method _generateLocalImagePath
+      @param {String} filename The name of the file
+      @return {String} A path to the local asset
+    */
+    _generateLocalImagePath: function(filename) {
+      var staticURL = this.props.staticURL || '';
+      var basePath = `${staticURL}/static/gui/build/app/assets/images/store/`;
+      return basePath + filename;
+    },
+
+    /**
+      Generate the URL to a icon for a charm or bundle
+
+      @method _generateIconPath
+      @param {String} entityName The name of the charm or bundle
+      @return {String} A URL to the entity icon
+    */
+    _generateIconPath: function(entityName) {
+      return `https://api.jujucharms.com/charmstore/v5/${entityName}/icon.svg`;
+    },
+
+    /**
+      Generate the URL to a diagram for a bundle
+
+      @method _generateDiagramPath
+      @param {String} entityName The name of the bundle
+      @return {String} A URL to the bundles diagram
+    */
+    _generateDiagramPath: function(entityName) {
+      return 'https://api.jujucharms.com/charmstore/v5/bundle/' +
+        entityName + '/diagram.svg';
+    },
+
+    /**
+      Generate the URL to a users page on the storefront
+
+      @method _generateUserPath
+      @param {String} username The name of the user
+      @return {String} A URL to the users page on storefront
+    */
+    _generateUserPath: function(username) {
+      return `https://jujucharms.com/u/${username}`;
+    },
+
+    /**
+      The content for the write you own section of the page
+
+      @method _writeOurOwnSection
+      @return {Object} The contents of the section
+    */
+    _writeOurOwnSection: function() {
+      var doc = 'https://www.jujucharms.com/docs/stable/authors-charm-writing';
+      return (<div className="row row--write-your-own">
+          <div className="inner-wrapper clearfix">
+              <div className="six-col">
+                  <h2>Write a charm and join the ecosystem</h2>
+                  <p>Creating new charms it easy. Charms can be written
+                  in your choice of language and adapting existing
+                  scripts is straightforward. You can keep new charms
+                  private, or share them back with the community.</p>
+                  <p>
+                  <a
+                    href={doc}>
+                    Learn more about writing charms&nbsp;&rsaquo;
+                  </a></p>
+              </div>
+          </div>
+      </div>);
+    },
+
+    /**
+      The content for the development tools section of the page
+
+      @method _developmentToolsSection
+      @return {Object} The contents of the section
+    */
+    _developmentToolsSection: function() {
+      return (<div className="row row--border-bottom">
+        <div className="inner-wrapper">
+            <h2>Development tools</h2>
+            <ul className="twelve-col no-bullets equal-height">
+                <li className="three-col featured-entity">
```

Comment:
I'd prefer if these list items were generated from a map of data that could be passed in.